File: output/ChatGPT/avatar/Python/C++/codeforces_544_B.cpp
#include <iostream>
using namespace std;

int main() {
    int n, k;
    cin >> n >> k;
    int x = 0;
    if (k <= max(n * n / 2, n * n - (n * n / 2))) {
        cout << "YES" << endl;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i % 2 == j % 2 && x < k) {
                    cout << "L";
                    x++;
                } else {
                    cout << "S";
                }
            }
            cout << endl;
        }
    } else {
        cout << "NO" << endl;
    }

    return 0;
}

File: output/CodeLlama/avatar/Java/Python/codeforces_544_B.py
import sys

n, count = map(int, sys.stdin.readline().split())

a = (n + 1) // 2
b = n // 2
max = a * a + b * b
if count > max:
    print('NO')
    exit()
print('YES')
for i in range(n):
    for j in range(n):
        if (i + j) % 2 == 0 and count > 0:
            print('L', end='')
            count -= 1
        else:
            print('S', end='')
    print()